### PR TITLE
If a user doesn't have access to a project, unset it to not reveal it

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

This is pretty ineffective queries-wise, but it does prevent users from ever seeing a project that's restricted to them (noticed during _break the release_ – ideally private projects should practically not exists for those who don't have access to them). Previously they could see it if:

- they had no project set and there were no non-private projects, so the private one got set as the starting `current_team` value
- they were removed from the project, so it was kept as the `current_team`

The question is, is this worth the queries?

### CODE REVIEW

**Comment by Twixes** [comment]
Hm, I think it may be worth handling this edge case imperfectly and avoid the need for the extra queries for now…

File: posthog/models/user.py [hunk 1]
```diff
@@ -142,8 +142,17 @@ def organization(self) -> Optional[Organization]:
...
-            self.current_team = self.organization.teams.order_by("access_control", "id").first()  # Prefer open projects
+            # If there's no current project, use the first one, with preference for open projects
+            self.current_team = self.organization.teams.order_by("access_control", "name").first()
+            if self.current_team.get_effective_membership_level(self) is None:
+                # Unset if the user does not have access to the autodetermined project
+                # In this case they'll just have to set it themselves
+                self.current_team = None
             self.save()
+        else:
+            if self.current_team.get_effective_membership_level(self) is None:
```
Comment (mariusandra) [inline posthog/models/user.py:153]: Theoretically, `self.current_team` can still be `None` here.

File: posthog/models/user.py [hunk 2]
```diff
@@ -142,8 +142,17 @@ def organization(self) -> Optional[Organization]:
...
     def team(self) -> Optional[Team]:
         if self.current_team is None and self.organization is not None:
-            self.current_team = self.organization.teams.order_by("access_control", "id").first()  # Prefer open projects
+            # If there's no current project, use the first one, with preference for open projects
+            self.current_team = self.organization.teams.order_by("access_control", "name").first()
+            if self.current_team.get_effective_membership_level(self) is None:
+                # Unset if the user does not have access to the autodetermined project
+                # In this case they'll just have to set it themselves
+                self.current_team = None
             self.save()
```
Comment (mariusandra) [inline posthog/models/user.py:151]: This seems like a needless save if `current_team` is reset to `None` with the `if` above.

**Review by mariusandra — COMMENTED** [review]
There's nothing wrong with extra queries, but I'm not sure what's the state of this PR after the last comment, if it should go in or not. Found some questions about the code as well.

**Comment by Twixes** [comment]
After thinking about this more I think that #7142 solves this better, as handling on the frontend avoids the extra queries (~5 more queries on every project-based request) and presentation of this state is key anyway.

**Pull request closed without merging**